src/System.Net.NetworkInformation/src/System/Net/NetworkInformation/NetworkInterfacePal.Linux.cs

```diff
@@ -0,0 +1,49 @@
+﻿namespace System.Net.NetworkInformation
+{
+    internal static class NetworkInterfacePal
+    {
+        /// Returns objects that describe the network interfaces on the local computer.
+        public static NetworkInterface[] GetAllNetworkInterfaces()
+        {
+            return LinuxNetworkInterface.GetLinuxNetworkInterfaces();
+        }
+
+        public static bool GetIsNetworkAvailable()
+        {
+            foreach (var ni in GetAllNetworkInterfaces())
+            {
+                if (ni.OperationalStatus == OperationalStatus.Up)
```

Comment:
Another place we might look to optimize in the future.  We're populating all of the network data for each network, but then we only care about a particular property.